commands/parse/parse_test.go

```diff
@@ -30,64 +29,79 @@ func TestParseConfig(t *testing.T) {
 
 	for _, testunit := range testTable {
 		t.Run(testunit.name, func(t *testing.T) {
-			exitCallCount := 0
+			exit := 0
 			cmd := NewParseCommand(func(int) {
-				exitCallCount++
+				exit++
 			})
-			cmd.Run(cmd, testunit.fileList)
-
-			if exitCallCount != 1 {
+			err := cmd.RunE(cmd, testunit.fileList)
+			if err != nil {
+				t.Errorf("problem running parse command in test: %w", err)
+			}
+			if exit != 0 {
 				t.Errorf(
-					"It should called one time but we have exit code: %v",
-					exitCallCount,
+					"The exit code should be 0 but got: %v",
+					exit,
 				)
 			}
 		})
 	}
 }
 
-func TestInputFlagforParseConfig(t *testing.T) {
-	testTable := []struct {
+func TestInputFlagForparseInput(t *testing.T) {
+	testunit := struct {
 		name     string
 		input    string
 		fileList []string
 	}{
-		{
-			name:     "valid input flag parse for terraform version 2",
-			input:    "hcl2",
-			fileList: []string{"testdata/terraform.tf"},
-		},
+		name:     "valid input flag parse for terraform version 2",
+		input:    "hcl2",
+		fileList: []string{"testdata/terraform.tf"},
 	}
-
-	for _, testunit := range testTable {
-		t.Run(testunit.name, func(t *testing.T) {
-			viper.Reset()
-			viper.Set("input", "hcl2")
-			exitCallCount := 0
-			expected := `
-			"for_each": "${data.consul_key_prefix.environment.var}"`
-			output := capturer.CaptureOutput(func() {
-				cmd := NewParseCommand(func(int) {
-					exitCallCount++
-				})
-
-				cmd.Run(cmd, testunit.fileList)
-			})
-			viper.Reset()
-			assert.Assert(t, is.Contains(output, expected))
-		})
+	t.Run(testunit.name, func(t *testing.T) {
+		expectedFile := "testdata/terraform.tf"
+		expected := `{
+	"data.consul_key_prefix.environment": {
+		"path": "apps/example/env"
+	},
+	"output.environment": {
+		"value": "${{\n    id           = aws_elastic_beanstalk_environment.example.id\n    vpc_settings = {\n      for s in aws_elastic_beanstalk_environment.example.all_settings :\n      s.name =\u003e s.value\n      if s.namespace == \"aws:ec2:vpc\"\n    }\n  }}"
+	},
+	"resource.aws_elastic_beanstalk_environment.example": {
+		"application": "testing",
+		"dynamic.setting": {
+			"content": {
+				"name": "${setting.key}",
+				"namespace": "aws:elasticbeanstalk:application:environment",
+				"value": "${setting.value}"
+			},
+			"for_each": "${data.consul_key_prefix.environment.var}"
+		},
+		"name": "test_environment",
+		"setting": {
+			"name": "MinSize",
+			"namespace": "aws:autoscaling:asg",
+			"value": "1"
+		}
 	}
+}`
+		viper.Reset()
+		viper.Set("input", testunit.input)
```

Comment:
I don't really like having to do this here. I think I'd rather see us pass `input` into GetConfigurations or some other means such that we're not needing to reset/set viper configs.

Reply:
Again, `GetConfigurations()` is not a part of this PR's context.

Reply:
Ok, we can clean it up in another PR.